server/commands/list_objects.go

```diff
@@ -49,7 +50,39 @@ func (q *ListObjectsQuery) Execute(ctx context.Context, req *openfgapb.ListObjec
 		return nil, serverErrors.NewInternalError("", err)
 	}
 
-	resultsChan := make(chan string, q.ListObjectsMaxResults)
+	targetObjectType := req.GetType()
+	targetRelation := req.GetRelation()
+
+	model, err := q.Datastore.ReadAuthorizationModel(ctx, req.GetStoreId(), req.GetAuthorizationModelId())
+	if err != nil {
+		if errors.Is(err, storage.ErrNotFound) {
+			return nil, serverErrors.TypeNotFound(targetObjectType)
+		}
+		return nil, err
+	}
+
+	typesys := typesystem.New(model)
+
+	_, ok := typesys.GetTypeDefinition(targetObjectType)
+	if !ok {
+		return nil, serverErrors.TypeNotFound(targetObjectType)
+	}
+
+	relation, ok := typesys.GetRelation(targetObjectType, targetRelation)
+	if !ok {
+		return nil, serverErrors.RelationNotFound(targetRelation, targetObjectType, nil)
+	}
+
+	hasTypeInfo, ok := typesys.HasTypeInfo(targetObjectType, targetRelation)
+	if !ok {
+		// todo(jon-whit): handle error
+	}
+
+	resultsChan := make(chan string, 1)
```

Comment:
Need a minor refactor here to avoid goroutine leaks. 

The `ConnectedObjects` API should create a top-level errgroup and set a concurrency limit on the group and then pass that errgroup down the call chain so that the total concurrency is limited instead of concurrency at each level of the recursive expansion.

Then this should be set to `resultsChan := make(chan string, maxConcurrency)` or whatever..

Reply:
Will save this for a later optimization outside of the scope of this PR, because this will require a lot of changes to the orchestration of this code.